File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
string = input()
M = len(string)

arr = string.split('-')
total = 0
nums = arr[0].split('+')
for num in nums:
    total += int(num)

for i in range(1, len(arr)):
    nums = arr[i].split('+')
    for num in nums:
        total -= int(num)

print(total)

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();

        String[] arr = input.split("-");
        int total = 0;

        String[] first = arr[0].split("\\+");
        for (String num : first) {
            total += Integer.parseInt(num);
        }

        for (int i = 1; i < arr.length; i++) {
            String[] nums = arr[i].split("\\+");
            for (String num : nums) {
                total -= Integer.parseInt(num);
            }
        }
        
        System.out.println(total);
    }
}
